omnichain/multioutput/contracts/MultiOutput.sol

```diff
@@ -74,21 +65,87 @@ contract MultiOutput is zContract, Ownable {
                 amountToTransfer += leftOver;
             }
 
-            uint256 outputAmount = SwapHelperLib._doSwap(
-                systemContract.wZetaContractAddress(),
-                systemContract.uniswapv2FactoryAddress(),
-                systemContract.uniswapv2Router02Address(),
-                zrc20,
-                amountToTransfer,
-                targetZRC20,
-                0
+            bytes memory recipient = abi.encodePacked(
+                BytesHelperLib.addressToBytes(evmRecipient)
             );
-            SwapHelperLib._doWithdrawal(
-                targetZRC20,
-                outputAmount,
-                BytesHelperLib.addressToBytes(recipient)
+            
+            if (targetZRC20 == BITCOIN_ZRC20_ADDRESS) {
+                if (btcRecipient.length == 0) revert InvalidRecipient();
+                recipient = abi.encodePacked(btcRecipient);
+            }
+
+            _doSwapAndWithdraw(zrc20, amountToTransfer, targetZRC20, recipient);
+        }
+    }
+
+    function _doSwapAndWithdraw(
+        address zrc20,
+        uint256 amountToTransfer,
+        address targetZRC20,
+        bytes memory recipient
+    ) internal {
+        uint256 outputAmount = SwapHelperLib._doSwap(
+            systemContract.wZetaContractAddress(),
+            systemContract.uniswapv2FactoryAddress(),
+            systemContract.uniswapv2Router02Address(),
+            zrc20,
+            amountToTransfer,
+            targetZRC20,
+            0
+        );
+        SwapHelperLib._doWithdrawal(
+            targetZRC20,
+            outputAmount,
+            bytes32(recipient)
```

Comment:
@lukema95 you're converting the recipient address to `bytes32`, but for Bitcoin the address is longer, so it doesn't fit into 32 bytes.

So this hex version of a Bitcoin address:

```
0x746231713873687a663761666333726877386e367736656333327338683665326d727730373764306767
```

Gets truncated to:

```
0x746231713873687a663761666333726877386e36773665633332733868366532
```

And this is passed to the withdraw function.

I propose you replace `_doWithdrawal` with the actual `.withdraw` (like in this swap tutorial), to avoid the `bytes32` issue.